ctx.go

```diff
@@ -260,33 +260,60 @@ func (c *Ctx) BaseURL() string {
 	return c.baseURI
 }
 
-// Body contains the raw body submitted in a POST request.
+// BodyRaw contains the raw body submitted in a POST request.
 // Returned value is only valid within the handler. Do not store any references.
 // Make copies or use the Immutable setting instead.
+func (c *Ctx) BodyRaw() []byte {
+	return c.fasthttp.Request.Body()
+}
+
+// Body contains the raw body submitted in a POST request.
+// This method will decompress the body if the 'Content-Encoding' header is provided.
+// It returns the original (or decompressed) body data which is valid only within the handler.
+// Don't store direct references to the returned data.
+// If you need to keep the body's data later, make a copy or use the Immutable option.
 func (c *Ctx) Body() []byte {
-	var err error
-	var encoding string
-	var body []byte
-	// faster than peek
-	c.Request().Header.VisitAll(func(key, value []byte) {
-		if c.app.getString(key) == HeaderContentEncoding {
-			encoding = c.app.getString(value)
+	var (
+		err                error
+		body, originalBody []byte
+		encodingOrder      = []string{"", "", ""}
+	)
+
+	// Split and get the encodings list, in order to attend the
+	// rule defined at: https://www.rfc-editor.org/rfc/rfc9110#section-8.4-5
+	encodingOrder = getSplicedStrList(c.Get(HeaderContentEncoding), encodingOrder)
+	if len(encodingOrder) == 0 {
+		return c.fasthttp.Request.Body()
+	}
+
+	for index, encoding := range encodingOrder {
+		switch encoding {
+		case StrGzip:
+			body, err = c.fasthttp.Request.BodyGunzip()
+		case StrBr, StrBrotli:
+			body, err = c.fasthttp.Request.BodyUnbrotli()
+		case StrDeflate:
+			body, err = c.fasthttp.Request.BodyInflate()
+		default:
+			return body
 		}
-	})
 
-	switch encoding {
-	case StrGzip:
-		body, err = c.fasthttp.Request.BodyGunzip()
-	case StrBr, StrBrotli:
-		body, err = c.fasthttp.Request.BodyUnbrotli()
-	case StrDeflate:
-		body, err = c.fasthttp.Request.BodyInflate()
-	default:
-		body = c.fasthttp.Request.Body()
+		if err != nil {
+			return []byte(err.Error())
+		}
+
+		if index < len(encodingOrder)-1 {
```

Comment:
Can you add a testcase for this usecase

Reply:
Done it, I also add some more testcases, for the error and others